spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java

```diff
@@ -252,6 +251,37 @@ private void deployStream(StreamDefinition stream, Map<String, String> cumulated
 		}
 	}
 
+	private ModuleType determineModuleType(ModuleDefinition moduleDefinition, int index, boolean hasNext) {
```

Comment:
I was about to file another comment when I realized `index` is in "reverse" order. May want to rename / add javadoc to clarify